modules/deployment/src/main/java/org/eclipse/xpanse/modules/deployment/DeployService.java

```diff
@@ -107,6 +108,11 @@ public DeployTask createNewDeployTask(DeployRequest deployRequest) {
             log.error("Service not accepted Eula.");
             throw new EulaNotAccepted("Service not accepted Eula.");
         }
+        if (!existingServiceTemplate.getOcl().getBilling().getBillingModes()
+                .contains(deployRequest.getBillingMode())) {
+            log.error("The service template does not support this billing mode.");
+            throw new BillingModeNotSupported("BillingMode not supported.");
+        }
```

Comment:
The error message should be updated in the same way above.

Reply:
Good advice.